tests/common.go

```diff
@@ -0,0 +1,226 @@
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"github.com/portworx/torpedo/drivers/node"
+	// import aws driver to invoke it's init
+	_ "github.com/portworx/torpedo/drivers/node/aws"
+	// import ssh driver to invoke it's init
+	_ "github.com/portworx/torpedo/drivers/node/ssh"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	// import k8s driver to invoke it's init
+	_ "github.com/portworx/torpedo/drivers/scheduler/k8s"
+	"github.com/portworx/torpedo/drivers/volume"
+	// import portworx driver to invoke it's init
+	_ "github.com/portworx/torpedo/drivers/volume/portworx"
+	"github.com/portworx/torpedo/pkg/log"
+)
+
+const (
+	// DefaultSpecsRoot specifies the default location of the base specs directory in the Torpedo container
+	DefaultSpecsRoot     = "/specs"
+	schedulerCliFlag     = "scheduler"
+	nodeDriverCliFlag    = "node-driver"
+	storageDriverCliFlag = "storage"
+	specDirCliFlag       = "spec-dir"
+)
+
+const (
+	defaultScheduler     = "k8s"
+	defaultNodeDriver    = "ssh"
+	defaultStorageDriver = "pxd"
+)
+
+var context = ginkgo.Context
+
+// Step is an alias for ginko "By" which represents a step in the spec
+var Step = ginkgo.By
+var expect = gomega.Expect
+var haveOccurred = gomega.HaveOccurred
+var beEmpty = gomega.BeEmpty
+
+// InitInstance is the ginkgo spec for initializing torpedo
+func InitInstance() {
+	var err error
+	err = Inst().S.Init(Inst().SpecDir, Inst().N.String())
+	expect(err).NotTo(haveOccurred())
+
+	err = Inst().V.Init(Inst().S.String())
+	expect(err).NotTo(haveOccurred())
+
+	err = Inst().N.Init(Inst().S.String())
+	expect(err).NotTo(haveOccurred())
+}
+
+// ValidateContext is the ginkgo spec for validating a scheduled context
+func ValidateContext(ctx *scheduler.Context) {
+	ginkgo.Describe(fmt.Sprintf("For validation of %s app", ctx.App.Key), func() {
```

Comment:
why not directly use Describe, Context, Step, Expect etc in common as well?

Reply:
golang doesn't allow dot imports in non *_test.go files. So have to use explicit `ginkgo.`